docs/usage/data-processing.md

```diff
@@ -0,0 +1,101 @@
+use of the Java data engine for compaction. Compaction in Java is much slower and more expensive, so this is not
+recommended.
+
+We intend to keep the configuration as simple as possible for aggregation and filtering, rather than introducing many
+operations and options. We are planning alternatives to apply other types of processing in DataFusion.
+
+## Types of processing
+
+### Aggregation
+
+There are many use cases where we want to aggregate rows where the keys are the same. We can configure this in the table
+property [`sleeper.table.aggregations`](properties/table/data_definition.md).
+
+For example, we have a three column table where the key field is a string called 'id', the first value field is a long
+called 'count' and the second value field is a long called 'last_seen'. If multiple rows with the same id are inserted
+then we want to add the counts and take the maximum of the values in the last_seen column. We can achieve this by
+setting `sleeper.table.aggregations` to `sum(count), max(last_seen)`.
+
+This means that every time a compaction or a query runs, rows will be combined together if they have the same value for
+all row keys and sort keys. This becomes part of the definition of the table, and you will only ever see the data with
+that process pre-applied.
+
+See the the [table properties documentation](properties/table/data_definition.md) for the supported aggregation
+operations. We also support this for value fields that contain nested map data. This can be applied with the Java or
+DataFusion data engine.
+
+### Filtering
+
+There are use cases that require filtering of data based on values of fields. We can configure this in the table
+property [`sleeper.table.filters`](properties/table/data_definition.md).
+
+For example, if the rows in our table have a long field called 'timestamp', we may want to remove a row whenever the
+timestamp is older than 2 weeks. We can achieve this by setting `sleeper.table.filters`
+to `ageOff(timestamp, 1209600000)`. The second parameter is the maximum age in milliseconds.
+
+This means that every time a compaction or a query runs, rows will be excluded from the output if they have a timestamp
+that's older than the specified age. This becomes part of the definition of the table, and you will never see data that
+has a timestamp older than that according to the clock of the query processor.
+
+See the the [table properties documentation](properties/table/data_definition.md) for the supported filtering
+operations. This can be applied with the Java or DataFusion data engine.
+
+### Custom iterators
+
+For arbitrary data processing, you can write your own iterator implementing the Java interface `ConfigStringIterator`.
+This lets you insert operations to be performed on rows as Sleeper reads the underlying data, which is usually done in
+parallel across many machines. Note that this will have an impact on startup time during queries, as your code will be
```

Comment:
Change this to "This lets you insert operations to be performed on rows as Sleeper reads the underlying data during compactions."